Show local weather

haszabaj opened the pull request.

haszabaj committed:
Add files via upload

haszabaj committed:
Add files via upload

lsiedlaczek commented on localweather.js:
```diff
@@ -0,0 +1,45 @@
+var wd;
+
+
+function switchUnit(currentTempF, c) {
```
jednak nazywanie zmiennych jednoliterowo jest slabe.

lsiedlaczek commented on localweather.js:
```diff
@@ -0,0 +1,45 @@
+var API_KEY = "b66507bb73d941dcbcc192051170507";
```
kazalem Ci zalatwic sobie ksiazke javascript design patterns stefanova. prosze przeczytaj tam o module pattern i zastosuj go w tym kodzie

lsiedlaczek replied:
jezeli pliku nie uzywamy to go usuwamy z repo

lsiedlaczek commented on localweather.js:
```diff
@@ -0,0 +1,45 @@
+    return currentTempF + '&deg;' + 'F';
+}
+
+function render(wd, cel) {
```
zmienne wd i cel nic nie mowia.

lsiedlaczek commented on localweather.js:
```diff
@@ -0,0 +1,45 @@
+    var loc;
+    $.getJSON('https://ipinfo.io', function(d) {
+        loc = d.loc.split(",");
+			$.getJSON('https://api.apixu.com/v1/current.json?key=' + API_KEY + '&q=' + loc[0] + ',' + loc[1], function(apiData) {
```
co to za wciecia kosmiczne?

lsiedlaczek commented on localweather.js:
```diff
@@ -0,0 +1,45 @@
+	$('#currentTemp').prepend('<img src="' + iconSrc + '" id="#icon">');
+}
+
+$(function() {
```
skoro zastosowalas czesciowy module pattern (z jquery), czemu nie wrzucilas funkcji powyzej do srodka?

lsiedlaczek commented on localweather.js:
```diff
@@ -0,0 +1,45 @@
+
+function switchUnit(currentTempF, c) {
+    if (c) {
+        var valueInC = (currentTempF - 32) * (5 / 9) + '&deg;' + 'C';
```
moglas zrobic funcje pomocnicza "toDegrees()", ktora bralaby wartosc numeryczna + stopien i laczyla to ladnie w string

lsiedlaczek commented on localweather.js:
```diff
@@ -0,0 +1,45 @@
+    $('#currentWeather').html(currentWeather);
+    $('#currentTemp').html(currentTempF);
+    $('#icon').html('https:' + icon);
+	$('#currentTemp').prepend('<img src="' + iconSrc + '" id="#icon">');
```
czemu wciecie? prosze zainstaluj sobie jakis beautifier / editorconfig albo jshint / jslint jako plugin do edytora.

haszabaj committed:
Add files via upload

lsiedlaczek commented on WeatherApp - updated.js:
```diff
@@ -0,0 +1,73 @@
+      },
+
+    changeTemp: function() {
+      WeatherApp.toggle.on("click", function() {
```
jakie WeatherApp?

lsiedlaczek commented on localweather.html:
```diff
@@ -0,0 +1,22 @@
+<html>
```
dlaczego w htmlu nigdzie nie ma odwolania ani do cssa ani do pliku js?

lsiedlaczek commented on WeatherApp - updated.js:
```diff
@@ -0,0 +1,73 @@
+
+var weatherApp = (function () {
```
chyba nie do konca jednak skumalas idee modulow. postaram sie opisac to na szybko tutaj. 

*po co uzywamy modulow*
1.  po to, zeby opakowac X funkcji w jedna zmienna globalna.
```javascript
// dlatego robimy:
var weatherApp = (function() {
  // funkcje, ktore staly sie metodami modulu
})();
```

2. po to, zeby podzielic funkcjonalnosci na prywatne i publiczne
dotyczy to bardzo architektury oprogramowania i ladnego, "zgodnego z litera prawa" pisania kodu. 
```javascript
// na przyklad modul time do zarzadzania czasem
var time = (function() {
  /*
    kod ktory chcielibysmy, zeby wykonywal sie zawsze przy deklaracji modulu 
    zazwyczaj chcesz tu miec tylko defaultowe zmienne konfiguracyjne
  */
  var current = new Date();

  // definiujemy co zwrocimy do publicznego API tego modulu
  // oznacza to, ze z tych akcji bedzie mozna korzystac poza modulem
  return {
    // nalezy pamietac, ze tutaj nie robimy wywolania funkcji. dlatego jest "spend" a nie "spend()"
    // robimy przypisanie do metody API o nazwie spend funkcji prywatnej spend
    spend: spend, 
    waste: waste
    // metoda spend nie znalazla sie w tym miejscu, 
    //oznacza to ze poza modulem nie bedzie mozna jej wywolac
  }
 
  // deklaracje funkcji sa hostowane na gore modulu. dzieki temu mozesz je umiescic pod returnem
  // i w pierwszej kolejnosci widziec API  jakie zwraca dany modul
  function spend() { ... }
  function waste() { ... }
  // nie chce dawac nikomu mozliwosci ustawiania czasu, wiec to bedzie metoda prywatna
  function set() { ... }
})();
```

3. po to zeby pogrupowac je po uzytecznosci
```javascript
// czyli mozemy stworzyc sobie pare modulow. mozemy stworzyc modul do zarzadzania czasem
// z poprzedniego przykladu, ktory ma w sobie tylko funkcje do zarzadzania czasem
// i stworzyc do niego modul space do zarzadzania przestrzenia 
// wpisuje sie to w zasady SOLID z programowania obiektowego:
// https://pl.wikipedia.org/wiki/SOLID_(programowanie_obiektowe)
var space = (function() {
  return {
     travel: travel
  }

  function travel() { ... }
})();
````
w Twojej apce nie ma czego dzielic, ale daje Ci pojecie jak to zastosowac ;)

*jak używamy modulu?*
1. deklarujemy moduly 
moduly najlepiej przechowywac w liczbie jeden per plik, odseparowane od innych rzeczy (typu kontrolery), ale poki co zrobmy wszystko w jednym pliku

2. tworzymy kontroler
kontroler moglby rowniez byc modulem, ktory po prostu mialby w sobie flow aplikacji i jej wywolanie.
```javascript
// tutaj podajemy nazwy modulow jak chcemy je widziec lokalnie
// my weatherApp zamienilismy sobie na weather w celach pokazowych
var controller = (function(weather, someOther, $) {
  // wywoluj akcje z modulow ktorych chcesz uzyc
  someOtherModule.doSomething()
  someOtherModule.doSomethingElse();

  // ustawiaj listenery
  $(document).ready(onStart);
  $(document).find('.refresh').on('click', onClick);

  function onStart() {
    // skorzystaj sobie luzno z modulu weather
    var currentWeather = weather.getWeather();
    // ustaw jakiegos diva na wartosc currentWeather
    // ... 
  }

  function onClick() {
    // strzelaj ogniem
  }

// tutaj podajemy nazwy modulow jakie chcemy uzywac
})(weatherApp, someOtherModule, $);

3. dodajemy jsy z modulami i kontrolerem jako <script> do htmla
w pierwszej kolejnosci moduly pozniej kontroler - w koncu bedzie z nich korzystac.

4. bangla

*zaleznosci pomiedzy modulami*
moduly moga miec do siebie roznoraki stosunek:
1. moga byc od siebie zalezne - jezeli jeden modul wykorzystuje drugi (wywoluje go w swoim ciele)
2. moga byc zupelnie niezalezne - (jezeli jeden modul nie ma zadnych wywolan kodu w drugim)

dependencje (wykorzystywanie przez jeden modul innego modulu) sa czesto sposobem odseparowania kodu dokladniejszej implementacji od pewnej warstwy abstrakcji. sa to tez powszechnie uzywane pluginy lub biblioteki. np. jquery jest dependencja Twojego kodu. wykorzystujesz go u siebie za kazdym razem gdy wolasz $ (nazwa modulu jquery) lub jquery.

haszabaj committed:
Add files via upload

haszabaj committed:
Delete WeatherApp - pytania.js

haszabaj committed:
Add files via upload

1) pisałeś mi, że this nie ma szansy działać o ile nie użyję new. czytałam na ten temat i nie do końca rozumiem dlaczego - jeśli definiuję objekt i jego properties tak jak w tym kodzie (bez używania new i object.prototype) to według różnych przewodników powinno to dzialać.  mam jednak trudność w dostępie do lokalnych zmiennych (może to głupie, jesli tak to przepraszam, ale ten projekt siedzi mi na głowie zdecydowanie za dlugo i wszystko mi się już miesza). otóż rozumiem to tak - wewnątrz modułu określam zmienne (np. loc) i określam funkcje. w przypadku tego API call rozbłam to na dwie części - locationCall używa getJSON w celu zdobycia IP użytkownika i obróbce go za pomocą split, żeby długość i szerokość geograficzna mogły być użyte we właściwym APICall. testowałam to wiele razy i z poziomu różnych funkcji mam dostęp do tych zmiennych lub nie i nie bardzo potrafię zrozumieć przyczynę. czy jeśli najpierw w module deklaruję psutą z

haszabaj commented:
wkleiłam to jako komentarz do dodanego pliku, nie widać go w całości więc wklejam jeszcze raz :)

1) pisałeś mi, że this nie ma szansy działać o ile nie użyję new. czytałam na ten temat i nie do końca rozumiem dlaczego - jeśli definiuję objekt i jego properties tak jak w tym kodzie (bez używania new i object.prototype) to według różnych przewodników powinno to dzialać.  mam jednak trudność w dostępie do lokalnych zmiennych (może to głupie, jesli tak to przepraszam, ale ten projekt siedzi mi na głowie zdecydowanie za dlugo i wszystko mi się już miesza). otóż rozumiem to tak - wewnątrz modułu określam zmienne (np. loc) i określam funkcje. w przypadku tego API call rozbiłam to na dwie części - locationCall używa getJSON w celu zdobycia IP użytkownika i obróbce go za pomocą split, żeby długość i szerokość geograficzna mogły być użyte we właściwym APICall. testowałam to wiele razy i z poziomu różnych funkcji mam dostęp do tych zmiennych lub nie i nie bardzo potrafię zrozumieć przyczynę. czy to loc w każdym przypadku to zawsze to samo loc, ktore jest okreslone w ?  jeszcze raz przepraszam, ale utknęlam na tym totlnie i nie wiem już czy to ja kompletnie źle rozumiem same założenia czy jest tu jakiś banalny błąd którego nie widzę.
2) czy sam revealing module tutaj jest dobrze zastosowany? w sensie sam schemat?
3) czy render jest tu sensowne? myslałam nad użyciem Mustache.js, ale ciężko mi to testować, skoro same dane z API nie działają 
4) co powinno tutaj być w return? bo żadnej z tych funckji nie chciałabym dawać do publicznego dostępu.
5) nie zwracaj uwagi na switchUnit, to nie jest ostateczna forma tej funkcji.
